src/bopforge/continuous_pipeline/__main__.py

```diff
@@ -235,9 +247,17 @@ def main(args=None) -> None:
         nargs="+",
         help="Included actions, default all actions",
     )
+    parser.add_argument(
+        "-m",
+        "--metadata",
+        type=lambda x: dict(eval(x)),
```

Comment:
Have to do something weird to automatically parse the dictionary from string, please let me know if there are better ways to do this.

Reply:
Yeah I can think of a couple ways.

1) You can use `type=json.loads` to avoid using `eval`.  

I think using `json.loads` would results in the same interface for users -- they would need to construct a python dictionary or json string on the CLI when they call the function. That might be a bit of a hassle for users who want to pass metadata, because they'd need to understand how to precisely construct dictionaries and would need to escape strings. There is an alternative way to do this that might be easier to type (the downside is we'd lose arbitrary nesting but I'm not sure that's super important)

2) We can treat any extra arguments as metadata. Like `python script.py --metadata foo=bar fiz=baz` would map to metadata of {'foo': 'bar', 'fiz': 'baz'}. There's an example of defining a custom kwargs parser like that here: https://sumit-ghosh.com/posts/parsing-dictionary-key-value-pairs-kwargs-argparse-python/ (but I would suggest we change argument name from kwargs to metadata)

I personally prefer option 2 because it'd be easier for users. And I don't think giving up arbitrary nesting would be a big downside.

Reply:
Thank you so much @JosephWagner! The second option is awesome! I learned a lot. I updated the code, let me know what you think :)